cmake/cpu_extension.cmake

```diff
@@ -0,0 +1,80 @@
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+include_directories("${CMAKE_SOURCE_DIR}/csrc")
+
+#
+# Check the compile flags
+#
+list(APPEND CXX_COMPILE_FLAGS 
+    "-fopenmp"
+    "-DVLLM_CPU_EXTENSION")
+
+execute_process(COMMAND cat /proc/cpuinfo
```

Comment:
I'm not sure if cross-compilation is a goal here but is it possible we might not want to solely rely on `/proc/cpuinfo` of the build machine? Perhaps adding a manual override option?

Reply:
Added an env ```VLLM_CPU_AVX512BF16``` for this purpose.